internal/controller/nopresource/nopresource_test.go

```diff
@@ -72,3 +75,79 @@ func TestObserve(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileLogic(t *testing.T) {
+
+	c := []v1alpha1.ResourceConditionAfter{
+		{Time: "10s", ConditionType: "Ready", ConditionStatus: "False"},
+		{Time: "5s", ConditionType: "Ready", ConditionStatus: "False"},
+		{Time: "7s", ConditionType: "Ready", ConditionStatus: "True"},
+		{Time: "5s", ConditionType: "Synced", ConditionStatus: "False"},
+		{Time: "10s", ConditionType: "Synced", ConditionStatus: "True"},
+		{Time: "2s", ConditionType: "Ready", ConditionStatus: "False"},
+	}
+
+	cases := map[string]struct {
+		reason            string
+		resourcecondition []v1alpha1.ResourceConditionAfter
+		elapsedtime       time.Duration
+		want              []int
+	}{
+		"EmptyReconcileArray": {
+			reason:            "Empty slice should be returned in case no conditions specified till given time elapsed.",
+			resourcecondition: c,
+			elapsedtime:       time.Duration(1*time.Second + 999*time.Millisecond),
+			want:              []int{},
+		},
+		"SingleTypeReconcile": {
+			reason:            "Slice with a single element should be returned when a single condition type has been specified.",
+			resourcecondition: c,
+			elapsedtime:       time.Duration(2 * time.Second),
+			want:              []int{5},
+		},
+		"NormalReconcileBehaviour": {
```

Comment:
probably another test case after 5 sec to test that the condition of Ready: false is still activated